CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,27 +26,69 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+
+namespace {
+    quint32 TIMEOUT = 120;
+    quint32 SLEEP_TIME = 2;
+}
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+    exchangeName_ = settings.exchangeName();
+    ipAddress_ = settings.ipAddress();
+    udpPort_ = settings.udpPort();
+    setupChannel();
 }
 
 UdpMessageForwarder::~UdpMessageForwarder()
 {
 }
 
+
+void UdpMessageForwarder::setupChannel()
+{
+    quint32 i = 0;
+    do {
+        if (i++) {
+            qWarning() << "UdpMessageForwarder: Attempting to reconnect";
+        }
+        try {
+            channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), udpPort_);
+        } catch (std::exception&) {
+            if (channel_ == NULL) {
+                if (i == (TIMEOUT / SLEEP_TIME)) {
+                    qWarning() << "UdpMessageForwarder timed out waiting for connection to broker";
+                    throw;
+                }
+                qWarning() << "UdpMessageForwarder: error creating channel";
+                QThread::sleep(SLEEP_TIME);
+            }
+            else {
+                throw;
+            }
+        }
+
+
+    } while (channel_ == NULL);
+    channel_->DeclareExchange(exchangeName_.toStdString(), AmqpClient::Channel::EXCHANGE_TYPE_FANOUT);
+}
```

Comment:
I think the big problem here is that, since hermes is a background process, we won't really have a way to indicate if some problem occurred. I don't think it makes sense in the program to create any pop ups, but if we can close the program with a bad exit code, then we can delegate that responsibility to another program (Domovoi).